File: src/dlp/ast/sentences/Block.ts

import {Sentence} from "./Sentence";
import {AbstractLocatable} from "../AbstractLocatable";

export abstract class Block extends AbstractLocatable implements Sentence{

    sentences: Sentence[];

    constructor(sentences: Sentence[], line: number, column: number) {
        super(line,column);
        this.sentences = sentences;
    }
}

File: src/ast/sentences/Block.java
package ast.sentences;

import ast.AbstractLocatable;
import java.util.List;

public abstract class Block extends AbstractLocatable implements Sentence {

    private List<Sentence> sentences;

    public Block(List<Sentence> sentences, int line, int column) {
        super(line, column);
        this.sentences = sentences;
    }

    public List<Sentence> getSentences() {
        return sentences;
    }
}
